### PR TITLE
support specifying multiple vault IDs for a playbook run

### PR DESCRIPTION
**Pull request by ryanpetrello**
see: https://github.com/ansible/awx/issues/352

this is the _first half_ of support, but represents the bulk of the work; after @AlanCoding's https://github.com/ansible/awx/pull/740 PR merges, I'll also need to add some additional code to support `ASK` in `passwords_needed_to_start` for multivault so you can specify them at launch time (so that the UI knows which vault ID passwords to prompt for).

The goal is for the UI to provide `ASK` multi vault as described here:

https://github.com/ansible/awx/issues/352#issuecomment-337995178

### CODE REVIEW

**Comment by ryanpetrello** [comment]
cc @alikins

File: awx/main/fields.py [hunk 1]
```diff
@@ -415,6 +415,13 @@ def from_db_value(self, value, expression, connection, context):
         return value
 
 
+@JSONSchemaField.format_checker.checks('vault_id')
+def format_vault_id(value):
+    if '@' in value:
+        raise jsonschema.exceptions.FormatError('@ is not an allowed character')
```
Comment (ryanpetrello) [inline awx/main/fields.py:421]: We're building the `--vault-id <id>@prompt` for them; we only want them to type in the `<id>` part, not `id@prompt` or `id@/some/file/path/`

File: awx/main/models/credential.py [hunk 2]
```diff
@@ -689,6 +689,16 @@ def vault(cls):
                 'type': 'string',
                 'secret': True,
                 'ask_at_runtime': True
+            }, {
+                'id': 'vault_id',
+                'label': 'Vault Identifier',
+                'type': 'string',
+                'format': 'vault_id',
+                'help_text': ('Specify an (optional) Vault ID. This is '
```
Comment (jakemcdermott) [inline awx/main/models/credential.py:697]: Adding the vault id as an optional per-credential input field makes sense to me - my understanding is that we'll be picking up which vault id's correspond to passwords that need prompting using the second formatting approach proposed here: https://github.com/ansible/awx/issues/352#issuecomment-337995178, which should be enough information to cue the UI on which vault identifiers correspond to the individual vault credentials that need password prompting.

Comment (ryanpetrello) [reply awx/main/models/credential.py:697]: @jakemcdermott yep, that's exactly right; I plan on having the launch endpoint expose e.g.,

```json
{
    "passwords_needed_to_start": [
        "vault_password.id1",
        "vault_password.id2"
    ]
}
```

**Review by matburt — APPROVED** [review]

**Pull request merged**